File: Exo2.3.c
#include <stdio.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>

int main() 
{
    int i;
    pid_t pid;

    for (i = 1; i <= 5; i++) 
    {
        pid = fork();

        if (pid < 0)
        {
            perror("Erreur de création de fork");
            return (1);
        } else if (pid == 0) 
        {
            // Code du fils
            printf("Je suis le fils %d\n", i);
            break;  //Arrêter la boucle pour que le fils ne crée pas d'autres fils
        }
        // Le père continue la boucle pour créer les autres fils
    }

    // Le père attend tous les fils pour éviter les zombies
    if (pid > 0) 
    {
        for (i = 0; i < 5; i++) 
        {
            wait(NULL);
        }
        printf("Je suis le père\n");
    }

    return (0);
}

File: Exo2.3.py
import os

for i in range(1, 6):
    pid = os.fork()
    
    if pid == 0:
        # Code du fils
        print(f"Je suis le fils {i}")
        os._exit(0)  # Terminer le fils pour que il ne crée pas d autres fils
        break  # Optionnel, car os._exit termine déjà le fils
    # Le père continue la boucle pour créer les autres fils

# Code du père
for _ in range(5):
    os.wait()  # Attendre tous les fils

print("Je suis le père")